### ISSUE TITLE
Add rotation & translation attack?

### ISSUE THREAD

**Issue by yenchenlin** [tensorflow/cleverhans#490]
As shown by some recent papers [1, 2], simple transformations are sufficient to fool neural network-based vision models. Notably, worst-of-10 attack described in [2] can serve as a good baseline for black-box attack.

Do you think it's appropriate for cleverhans to include these transformation-based attacks?

[Here](http://nbviewer.jupyter.org/github/yenchenlin/cleverhans/blob/transformation-attack/cleverhans_tutorials/mnist_tutorial_tran.ipynb) is an ipython notebook demonstrating the effectiveness of these attack. I've tested the model in `mnist_tutorial_tf.py` and find that:

- accuracy on legitimate examples: **0.9928**
- accuracy on transformed examples: **0.5715**

where the transformation consists of (i) 30-degree rotation and (ii) 10% translation along each direction.

<img width="443" alt="screen shot 2018-08-04 at 8 42 43 am" src="https://user-images.githubusercontent.com/7057863/43670761-6aa4af66-97c2-11e8-8720-ce4412ee1959.png">

[1] A Rotation and a Translation Suffice: Fooling CNNs with Simple Transformations, Engstrom et al.
[2] Spatially Transformed Adversarial Examples, Xiao et al.

**Comment by goodfeli** [comment]
Yes, I think that would be great to add.

**Comment by yenchenlin** [comment]
@npapernot Yes, working on it!

**Comment by nottombrown** [comment]
I have a working implementation of this that matches the performance from the paper on ImageNet. Happy to contribute it if @yenchenlin 's implementation isn't ready yet

**Comment by yenchenlin** [comment]
Hello @nottombrown , Thanks!!! But I'm almost there, would be awesome if you can give me a hand in the PR.

**Comment by yenchenlin** [comment]
@nottombrown We can close this now?

**Comment by nottombrown** [comment]
Closed via #594 !
